### PR TITLE
add desired replication factor option

### PR DESCRIPTION
**Pull request by klorand**
for situation in which you want to increase replication factor on a topic, besides moving partitions amond nodes

### CODE REVIEW

File: src/main/java/siftscience/kafka/tools/KafkaTopicAssigner.java [hunk 1]
```diff
@@ -35,23 +35,26 @@ public KafkaTopicAssigner() {
...
+        int replicationFactor = desiredReplicationFactor;
         Set<Integer> partitions = Sets.newTreeSet();
         for (Map.Entry<Integer, List<Integer>> entry : currentAssignment.entrySet()) {
             int partition = entry.getKey();
             List<Integer> replicas = entry.getValue();
             partitions.add(partition);
             if (replicationFactor < 0) {
                 replicationFactor = replicas.size();
-            } else {
+            } else if (desiredReplicationFactor<0) {
```
Comment (kanakb) [inline src/main/java/siftscience/kafka/tools/KafkaTopicAssigner.java]: style nit: please add spaces on either side of the operator, i.e.

``` java
} else if (desiredReplicationFactor < 0) {
```

**Review by kanakb — CHANGES_REQUESTED** [review]

**Comment by kanakb** [comment]
Thanks for the contribution!

**Review by kanakb — APPROVED** [review]

**Comment by glasser** [comment]
This change (and this project in general) is very helpful. Would you consider uploading a new tarball release?

**Comment by stalaie** [comment]
@glasser, as @kanakb mentioned, thanks for your contribution. Working with the team here to address your question above. Stay tuned!

**Comment by glasser** [comment]
Cool! FWIW I found #6 to be necessary as well.

It's hard for me to believe that this isn't just how kafka-reassign-partitions.sh works by default!

**Comment by ekoontz** [comment]
Hi @glasser, thanks for your interest in kafka-assigner! I have created a new release 1.1, which includes this PR.
-Eugene

**Comment by ekoontz** [comment]
I will also release a 1.2 which includes #6.

**Pull request merged**